### ISSUE TITLE
encoding on windows

### ISSUE THREAD

**Issue by sckott** [sckott/cowsay#9]
Hey @noamross @richfitz @trinker @leeper 

We're having problems with this pkg on windows. E.g., 

![screen shot 2014-09-18 at 8 58 48 am](https://cloud.githubusercontent.com/assets/577668/4322959/c5b3c4aa-3f4c-11e4-9ff2-0096c97db2e4.png)

But on unix does

``` r
say("asdfsf", "longcat", length = 4)

 ----- 
 asdfsf 
 ------ 
    \   
     \
    .ﾊ,,ﾊ
    ( ﾟωﾟ)
    |つ  つ
    |    |
    |    |
    |    |
    |    |
    U " U
```

This is what longcat is in the pkg:

``` r
  longcat <-
    '
\n ----- \n %s \n ------ \n    \\\   \n     \\\
    .\uFF8A,,\uFF8A
    ( \uFF9F\u03C9\uFF9F)
    |\u3064  \u3064
%s
    U " U
    '
```

Any way to fix this to work on windows?

**Comment by ateucher** [comment]
I think the pertinent information is in here: http://cran.cnr.berkeley.edu/doc/manuals/r-devel/R-ints.html#Encodings-for-CHARSXPs. 

Basically it seems like, on Windows, functions which use C code like `message` and `warning` can't properly deal with UTF-8 strings due to the Windows C compiler. It looks like `cat` and `print` have some sort of escape which gets around this... which is consistent with what @leeper observed (I experienced the same behaviour).

![capture](https://cloud.githubusercontent.com/assets/2816635/6085239/ecd10c1e-adec-11e4-823e-dff5099f6e40.PNG)

I played around with `iconv` a bit to no avail. I don't think you can catch and modify a message or warning after it's called and before it's printed?

**Comment by sckott** [comment]
@ateucher what if you use `type="string"`, then call `message()`, etc. manually after it comes back? does that work okay?

**Comment by ateucher** [comment]
Nope. I think as soon as `message`/`warning` get hold of it, it kills it:
![capture](https://cloud.githubusercontent.com/assets/2816635/6086029/7e48427e-adf3-11e4-8ed7-ec16ebfbbe47.PNG)

**Comment by sckott** [comment]
okay, boo, well, not sure what to do then. 

could just not use unicode characters and use the closest ascii characters...but you've kind of solved this another way by not using those that have unicode on windows

**Comment by ateucher** [comment]
Yeah, could do `gsub` on the \u strings only if os == windows.  But I wouldn't want to do it wholesale and make them look worse for everybody else.  Some of those might be hard to find decent replacements for though!

**Comment by sckott** [comment]
i think maybe we should just go with the solution you've already done - and push to CRAN!

**Comment by ateucher** [comment]
Agreed! It does function on Windows, so we should go for it.

**Comment by ateucher** [comment]
So I guess this isn't really fixed yet - the appveyor build is failing on the examples using non-ascii characters (`anxiouscat`). Would you want to put those in a `dontrun` section?

**Comment by sckott** [comment]
@ateucher  it seems like its an issue with the `match.arg` https://ci.appveyor.com/project/sckott/cowsay/build/1.0.218#L366 or am i looking at it wrong

**Comment by ateucher** [comment]
It is, but because `get_who()` removes anxiouscat etc from the vector whose names are used in `match.arg`.

**Comment by sckott** [comment]
I'll just remove the examples that don't work on windows...

**Comment by ateucher** [comment]
Yeah, good idea. That's the simplest solution
